### DISCUSSION TITLE
Can you somehow take two bitmaps by awaiting the second one?

### DISCUSSION THREAD

**Discussion by Veeksi** [PatilShreyas/Capturable#102] [Q&A]
Currently I want to implement custom theme switching. When I click a button I want to generate bitmap of original screen and basically show it instantly on top of everything. During the next frame the theme switches and I would like to capture second bitmap there.

But can I do that in the same onClick callback?

I tried with `rememberCoroutineScope` but capture method is documented like below and I don't think it's correct thing to do so?
> Make sure to call this method as a part of callback function and not as a part of the Composable function itself.

If I do like this:
```
scope.launch {
  captureController.capture()
  themeViewModel.setDarkThemeEnabled(!darkThemeEnabled)
  // Wait until the next frame to capture the image after the theme change.
  delay(50)
  captureController.capture()
}
```
I see that I capture both light mode and dark mode bitmaps, but I get recomposition problem where my texts do not switch to the correct color. 

That can be seen in the below image where I switched from dark to light mode and texts show correctly in bitmap images but not in the light mode.

<img src="https://github.com/PatilShreyas/Capturable/assets/55748788/a13ca197-9817-415b-9ed2-2627e626b145" width="256">

**Comment by faraz152** [comment]
yeah you can do both in the same coroutine, just await them in sequence. after the first capture switch the theme and wait a frame before capturing again:

```kotlin
scope.launch {
    val original = captureController.captureAsync().await()
    showOverlay(original)

    isDarkTheme = !isDarkTheme
    withFrameNanos { } // wait for compose to render the new theme
    val themed = captureController.captureAsync().await()
}
```
